### PR TITLE
Added composer.json support

### LINKED ISSUE #1: Add composer.json and such
- composer metadata
- autoload src directory

### PR DESCRIPTION
**Pull request by vinkla**
This pull request adds Composer support to the autoloader. Couple of questions before this pull request can be merged:

1. Should it be named `roots/bedrock-autoloader` or `roots/autoloader`?
2. Should it live on as a `mu-plugin` or as a "normal" Composer package (discussed in #5)?
3. Whom should we specify as authors?

Closes #1

### CODE REVIEW

**Comment by austinpray** [comment]
> * Should it be named `roots/bedrock-autoloader` or `roots/autoloader`?

I suppose if we are doing things properly this is not necessarily coupled with bedrock specifically? It should really be like "plugin-autoloader" or something. cc @roots/everyone 

> * Should it live on as a `mu-plugin` or as a "normal" Composer package (discussed in #5)?

Normal composer package as per @aaemnnosttv's recommendation. This is a library that one can use in their own file dropped in the mu-plugins directory. 

> * Whom should we specify as authors?

Added myself cause I'll be an admin on the composer repo

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      "homepage": "https://github.com/swalkinshaw"
+    },
+    {
+      "name": "Austin Pray",
+      "email": "austin@austinpray.com",
+      "homepage": "https://github.com/austinpray"
+    }
+  ],
+  "keywords": [
+    "autoloader", "bedrock", "mu-plugin", "must-user", "plugin", "wordpress"
```
Comment (austinpray) [inline composer.json]:
```suggestion
    "autoloader", "bedrock", "mu-plugin", "must-use", "plugin", "wordpress"
```

**Comment by retlehs** [comment]
@Foxaii is the original author, please add him to the list (at the :top:)

**Comment by vinkla** [comment]
Closing due to inactivity.

**Pull request closed without merging**